animal-options.lua

```diff
@@ -0,0 +1,175 @@
+--@ module = true
+
+local overlay = require('plugins.overlay')
+local widgets = require('gui.widgets')
+local gui = require('gui')
+
+local YES = 'Yes'
+local NO = 'No'
+
+-- Make sure an animal unit of your civ is selected as well as not a pet
+local function check_valid_unit()
+    local unit = dfhack.gui.getSelectedUnit(true)
+    return unit
+        and dfhack.units.isFortControlled(unit)
+        and dfhack.units.isAlive(unit)
+        and not dfhack.units.isPet(unit)
+        and dfhack.units.isAnimal(unit)
+end
+
+local function is_geldable()
+    local unit = dfhack.gui.getSelectedUnit(true)
+
+    -- If the unit is not male or already gelded do nothing
+    if unit.sex ~= 1 or unit.flags3.gelded == true then
+        return false
+    else
+        return true
+    end
+end
+
+creature_screen=defclass(creature_screen, overlay.OverlayWidget)
```

Comment:
"screen" is something specific in DF (an element on the viewscreen stack), and this is not it. `AnimalActionsWidget` might be more appropriate (also notice PascalCase class naming conventions)

Reply:
Switched `creature_screen` to suggested `AnimalActionsWidget`